actions/admin/settings/180.dkim.php

```diff
@@ -36,25 +36,25 @@
 					'varname' => 'dkim_prefix',
 					'type' => 'string',
 					'string_type' => 'dir',
-					'default' => '/etc/postfix/dkim/',
+					'default' => '/etc/dkim-froxlor/',
 					'save_method' => 'storeSettingField'
 				),
-				'dkim_domains' => array(
+/*				'dkim_domains' => array(
 					'label' => $lng['dkim']['dkim_domains'],
 					'settinggroup' => 'dkim',
 					'varname' => 'dkim_domains',
 					'type' => 'string',
 					'string_regexp' => '/^[a-z0-9\._]+$/i',
 					'default' => 'domains',
 					'save_method' => 'storeSettingField'
-				),
+				), */
 				'dkim_dkimkeys' => array(
 					'label' => $lng['dkim']['dkim_dkimkeys'],
 					'settinggroup' => 'dkim',
 					'varname' => 'dkim_dkimkeys',
 					'type' => 'string',
-					'string_regexp' => '/^[a-z0-9\._]+$/i',
-					'default' => 'dkim-keys.conf',
+					'string_type' => 'dir',				// DF8OE
+					'default' => '/etc/dkim_keys_froxlor/',
```

Comment:
why not use /etc/dkim/froxlor/{conf,keys} for dkim.dkim_prefix and dkim.dkim_dkimkeys (we could get rid of some 'dkims' here, like dkim.prefix and dkim.keys for example)

Reply:
I do not understand what you are meaning...

Reply:
put `/etc/dkim-froxlor/` and `/etc/dkim_keys_froxlor/` in one parent directory -> `/etc/dkim/froxlor/` and in there have folders `conf/config` and  keys`.

Second part I was talking about the settings names, like `dkim.dkim_dkimkeys` - three dkims' in there :P we sould change that to just `dkim.keys_path` or similiar

Reply:
We talked about his yesterday and you said you wanted NO SYMLINKS. So it is mandatory that config files will be placed in milters config directory - but not keys. I have solved that using ONE directory and symlinks - but you did not want that. So I splitted up. Regarding naming: yes - too much dkim ;) But renaming will also affect names in tables - I did not want to touch this...

Reply:
I don't see any symlinks there. besides, these are just the DEFAULTS, of course amavis wants its config in `/etc/amavis/conf.d/` - that's why this is a setting

Reply:
Misunderstanding I think. You are talking about the DEFAULTS - I understood "put config and keys in one folder" ;)
I was not sure about the defaults because they should differ for amavis <> OpenDKIM and I found no way to switch the defaults by service type...

Reply:
```php
'default' => (\Froxlor\Settings::Get('dkim.dkim_servicetype') == 0 ? '/etc/amavis/conf.d/' : '/etc/opendkim/'),
```

for example :)

Reply:
perfect.

Reply:
I have not changed the double named dkim_dkim.... because of it affects database tables, too. I do not want to run into trouble when I change now and upgrade froxlor later.

Reply:
database updates for this feature are missing completely anyway and if done correctly you won't have any troubles upgrading froxlor later